becquerel/core/rebin.py

```diff
@@ -255,29 +216,38 @@ def _rebin_interpolation(in_spectrum, in_edges, out_edges, slopes):
                 high = min(in_right_edge, out_right_edge)
             # Calc counts for this bin
             out_spectrum[out_idx] += _counts(slope, offset, low, high)
-    return out_spectrum
 
 
-@nb.jit(nb.i8[:](nb.i8[:], nb.f8[:], nb.f8[:], nb.f8[:]),
-        locals={'energies': nb.f8[:], 'in_idx': nb.u4, 'energy_idx': nb.u4},
-        nopython=True)
-def _rebin_listmode(in_spectrum, in_edges, out_edges, slopes):
-    """
-    Stochastic rebinning method: spectrum-histogram to listmode then back
+@nb.guvectorize([(nb.i8[:], nb.f8[:], nb.f8[:], nb.i8[:])],
+                "(n),(N),(m)->(m)")
+def _rebin_listmode(
+        in_spectrum, in_edges, out_edges_no_rightmost, out_spectrum):
+    """Stochastic rebinning method: spectrum-histogram to listmode then back.
+
+    rightmost edge in the parameter out_edges chopped off, in order for
+    nb.guvectorize to work, because it requires the output dimensions
+    to be specified by the dimensions of the input parameters
+    (i.e. m -> m-1 is not allowed).
+    This works since we put all overflow values in the leftmost and rightmost
+    bins anyways
 
     TODO Assume piecewise constant (ie steps, flat within each bin)
          distribution for in_spectrum (for now). slopes is unused.
 
     Args:
       in_spectrum: iterable of input spectrum counts in each bin
       in_edges: iterable of input bin edges (len = len(in_spectrum) + 1)
-      out_edges: iterable of output bin edges
-      slopes: unused, just for keeping number of arguments the same as
-              _rebin_interpolation()
+      out_edges_no_rightmost: iterable of output bin edges
+                              (sans the rightmost bin)
+      out_spectrum: for nb.guvectorize; do not actually give this as an arg
```

Comment:
Can you clarify:

> do not actually give this as an arg

Reply:
Also, add a note that this is the same length as `out_edges_no_rightmost`

Reply:
Done